test/blackbox/common/RTPSBlackboxTestsPersistence.cpp

```diff
@@ -190,8 +189,6 @@ TEST_P(Persistence, RTPSAsNonReliableWithPersistence)
 
     // Discover, send and receive
     run_one_send_recv_test(reader, writer, 13, false);
-    reader.destroy();
-    writer.destroy();
```

Comment:
Why are these being removed?

Reply:
Those methods are already called within the previous method `run_one_send_recv_test` method. In fact, some of the tests called these methods after finishing `run_one_send_recv_test` while others just omitted them. I removed them to unify the test's logic but maybe it is better to remove the destruction calls from the `run_one_send_recv_test` method and include them in the general test. In this case, we should add it after each round (`run_one_send_recv_test` is called twice in several tests)